database/mongo.go

```diff
@@ -0,0 +1,48 @@
+package database
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/joho/godotenv"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func ConnectToDB() (*mongo.Database, error) {
+	dbName := "api_test"
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	uri, err := getDotEnvVariable("DB_URI")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
General advice: don't do anything between the initialization of timing-out context and where you use it. The clock is ticking, tick-tock...